tools/rvn/CommandLineApp.cs

```diff
@@ -51,6 +61,137 @@ public static int Run(string[] args)
             }
         }
 
+        private static void ConfigureSetupPackage()
+        {
+            _app.Command("create-setup-package", cmd =>
+            {
+                cmd.Description = "This command creates RavenDB setup ZIP file";
+                cmd.HelpOption(HelpOptionString);
+                
+                var mode = ConfigureModeOption(cmd);
+                var setupParam = ConfigureSetupParameters(cmd);
+                var packageOutPath = ConfigurePackageOutputFile(cmd);
+                var certPath = ConfigureCertPath(cmd);
+                var certPass = ConfigureCertPassword(cmd);
+
+                cmd.OnExecuteAsync(async token =>
+                {
+                    var modeVal = mode.Value();
+                    var setupParamVal = setupParam.Value();
+                    var packageOutPathVal = packageOutPath.Value();
+                    var certPathVal = certPath.Value();
+                    var certPassTuple = (certPass.Value(), Environment.GetEnvironmentVariable("RVN_CERT_PASS"));
+                    
+                    return await CreateSetupPackage(new CreateSetupPackageParameters
+                    {
+                        SetupInfoPath = setupParamVal,
+                        PackageOutputPath = packageOutPathVal,
+                        Command = cmd,
+                        SetupMode = modeVal,
+                        CertificatePath = certPathVal,
+                        CertPassword = certPassTuple,
+                        Progress = new SetupProgressAndResult(tuple =>
+                        {
+                            if (tuple.Message != null)
+                            {
+                                Console.WriteLine(tuple.Message);
+                            }
+
+                            if (tuple.Exception != null)
+                            {
+                                Console.Error.WriteLine(tuple.Exception.Message);
+                            }
+                        }),
+                        CancellationToken = token
+                    });
+                });
+            });
+        }
+        
+        private static async Task<int> CreateSetupPackage(CreateSetupPackageParameters parameters)
+        {
+            byte[] zipFile;
+            SetupInfo setupInfo;
+
+            if (File.Exists(parameters.SetupInfoPath) == false)
```

Comment:
Let's move all the validation of parameters to earliest executed piece possible.